File: src/frontend/src/data/entities/connector/categories/ConnectorCategories.ts
import { ArchiveConnectorCategory } from "./ArchiveConnectorCategory";
import { ConnectorCategoriesCatalog } from "./ConnectorCategoriesCatalog";
import { RepositoryConnectorCategory } from "./RepositoryConnectorCategory";

export function registerConnectorCategories(): void {
    // New categories must be registered here
    ConnectorCategoriesCatalog.registerItem(RepositoryConnectorCategory.CategoryID, new RepositoryConnectorCategory());
    ConnectorCategoriesCatalog.registerItem(ArchiveConnectorCategory.CategoryID, new ArchiveConnectorCategory());
}

File: src/connectors/base/data/entities/connector/categories/connector_categories.py
from .connector_categories_catalog import ConnectorCategoriesCatalog


def register_connector_categories() -> None:
    from .archive_connector_category import ArchiveConnectorCategory
    from .repository_connector_category import RepositoryConnectorCategory

    # New categories must be registered here
    ConnectorCategoriesCatalog.register_item(
        ArchiveConnectorCategory.category_id, ArchiveConnectorCategory()
    )
    ConnectorCategoriesCatalog.register_item(
        RepositoryConnectorCategory.category_id, RepositoryConnectorCategory()
    )
